File: data/snippets/quiz018.cpp
class Report {
private:
    std::string title;
    std::string content;
    
public:
    Report(const std::string& title, const std::string& content) 
        : title(title), content(content) {}
        
    std::string getTitle() const { return title; }
    std::string getContent() const { return content; }
};

class ReportData {
private:
    std::vector<double> values;
    std::vector<std::string> labels;
    std::string dataSource;
    
public:
    void addValue(double value) { values.push_back(value); }
    void addLabel(const std::string& label) { labels.push_back(label); }
    void setDataSource(const std::string& source) { dataSource = source; }
    
    std::vector<double> getValues() const { return values; }
    std::vector<std::string> getLabels() const { return labels; }
    std::string getDataSource() const { return dataSource; }
    int getDataCount() const { return values.size(); }
};

class ReportFormatter {
public:
    // 这个方法对ReportData类的数据特别感兴趣
    std::string formatChartData(const ReportData& data) {
        std::stringstream result;
        
        // 访问大量ReportData的数据和方法
        result << "Chart Data (from " << data.getDataSource() << "):\n";
        result << "Total data points: " << data.getDataCount() << "\n";
        
        auto values = data.getValues();
        auto labels = data.getLabels();
        
        for (size_t i = 0; i < values.size() && i < labels.size(); ++i) {
            result << labels[i] << ": " << values[i] << "\n";
        }
        
        // 计算一些统计数据
        double sum = 0;
        for (double value : values) {
            sum += value;
        }
        double average = values.empty() ? 0 : sum / values.size();
        
        result << "Average: " << average << "\n";
        
        return result.str();
    }
    
    // 这个方法也过度使用ReportData的数据
    std::string generateSummary(const ReportData& data) {
        std::stringstream result;
        result << "Summary from " << data.getDataSource() << ":\n";
        
        auto values = data.getValues();
        
        if (values.empty()) {
            result << "No data available.";
            return result.str();
        }
        
        // 找出最大值和最小值
        double max = values[0];
        double min = values[0];
        
        for (double value : values) {
            if (value > max) max = value;
            if (value < min) min = value;
        }
        
        result << "Range: " << min << " to " << max << "\n";
        result << "Number of data points: " << data.getDataCount() << "\n";
        
        return result.str();
    }
    
    std::string formatReport(const Report& report) {
        return "Report: " + report.getTitle() + "\n\n" + report.getContent();
    }
};

File: data/snippets/quiz018.java
import java.util.List;
import java.util.ArrayList;

class Report {
    private String title;
    private String content;
    
    public Report(String title, String content) {
        this.title = title;
        this.content = content;
    }
    
    public String getTitle() { return title; }
    public String getContent() { return content; }
}

class ReportData {
    private List<Double> values;
    private List<String> labels;
    private String dataSource;
    
    public ReportData() {
        this.values = new ArrayList<>();
        this.labels = new ArrayList<>();
    }
    
    public void addValue(double value) { values.add(value); }
    public void addLabel(String label) { labels.add(label); }
    public void setDataSource(String source) { dataSource = source; }
    
    public List<Double> getValues() { return values; }
    public List<String> getLabels() { return labels; }
    public String getDataSource() { return dataSource; }
    public int getDataCount() { return values.size(); }
}

class ReportFormatter {
    
    public String formatChartData(ReportData data) {
        StringBuilder result = new StringBuilder();
        
        result.append("Chart Data (from ").append(data.getDataSource()).append("):\n");
        result.append("Total data points: ").append(data.getDataCount()).append("\n");
        
        List<Double> values = data.getValues();
        List<String> labels = data.getLabels();
        
        for (int i = 0; i < values.size() && i < labels.size(); i++) {
            result.append(labels.get(i)).append(": ").append(values.get(i)).append("\n");
        }
        
        double sum = 0;
        for (double value : values) {
            sum += value;
        }
        double average = values.isEmpty() ? 0 : sum / values.size();
        result.append("Average: ").append(average).append("\n");
        
        return result.toString();
    }
    
    public String generateSummary(ReportData data) {
        StringBuilder result = new StringBuilder();
        result.append("Summary from ").append(data.getDataSource()).append(":\n");
        
        List<Double> values = data.getValues();
        
        if (values.isEmpty()) {
            result.append("No data available.");
            return result.toString();
        }
        
        double max = values.get(0);
        double min = values.get(0);
        
        for (double value : values) {
            if (value > max) max = value;
            if (value < min) min = value;
        }
        
        result.append("Range: ").append(min).append(" to ").append(max).append("\n");
        result.append("Number of data points: ").append(data.getDataCount()).append("\n");
        
        return result.toString();
    }
    
    public String formatReport(Report report) {
        return "Report: " + report.getTitle() + "\n\n" + report.getContent();
    }
}
